### PR TITLE
Part of #12166: remove switch-case in bolliger_mabillard_track_flat

### PR DESCRIPTION
**Pull request by frutiemax**

### CODE REVIEW

File: src/openrct2/ride/coaster/BolligerMabillardTrack.cpp [hunk 1]
```diff
@@ -27,45 +27,21 @@ void bolliger_mabillard_track_flat(
...
-            case 2:
-                sub_98197C_rotated(
-                    session, direction, session->TrackColours[SCHEME_TRACK] | 17488, 0, 0, 32, 20, 3, height, 0, 6, height);
-                break;
-            case 3:
-                sub_98197C_rotated(
-                    session, direction, session->TrackColours[SCHEME_TRACK] | 17489, 0, 0, 32, 20, 3, height, 0, 6, height);
-                break;
-        }
+        uint32_t imageIds[] = { session->TrackColours[SCHEME_TRACK] | 17486, session->TrackColours[SCHEME_TRACK] | 17487,
```
Comment (Gymnasiast) [inline src/openrct2/ride/coaster/BolligerMabillardTrack.cpp:30]: Take out `session->TrackColours[SCHEME_TRACK] | ` and move that to the `sub_98197C_rotated()` call.

Comment (frutiemax) [reply src/openrct2/ride/coaster/BolligerMabillardTrack.cpp]: Oops this is called in a lot of other places...

File: src/openrct2/ride/coaster/BolligerMabillardTrack.cpp [hunk 2]
```diff
@@ -27,45 +27,24 @@ void bolliger_mabillard_track_flat(
...
-            case 2:
-                sub_98197C_rotated(
-                    session, direction, session->TrackColours[SCHEME_TRACK] | 17488, 0, 0, 32, 20, 3, height, 0, 6, height);
-                break;
-            case 3:
-                sub_98197C_rotated(
-                    session, direction, session->TrackColours[SCHEME_TRACK] | 17489, 0, 0, 32, 20, 3, height, 0, 6, height);
-                break;
-        }
+        uint32_t imageIds[] = { 17486, 17487, 17488, 17489 };
```
Comment (tupaschoal) [inline src/openrct2/ride/coaster/BolligerMabillardTrack.cpp:30]:
```suggestion
        std::array<uint32_t, NumOrthogonalDirections>imageIds = { 17486, 17487, 17488, 17489 };
```

File: src/openrct2/ride/coaster/BolligerMabillardTrack.cpp [hunk 3]
```diff
@@ -27,45 +27,24 @@ void bolliger_mabillard_track_flat(
...
-                sub_98197C_rotated(
-                    session, direction, session->TrackColours[SCHEME_TRACK] | 17146, 0, 0, 32, 20, 3, height, 0, 6, height);
-                break;
-            case 1:
-            case 3:
-                sub_98197C_rotated(
-                    session, direction, session->TrackColours[SCHEME_TRACK] | 17147, 0, 0, 32, 20, 3, height, 0, 6, height);
-                break;
-        }
+        uint32_t imageIDs[] = { 17146, 17147, 17146, 17147 };
```
Comment (tupaschoal) [inline src/openrct2/ride/coaster/BolligerMabillardTrack.cpp:42]:
```suggestion
        std::array<uint32_t, NumOrthogonalDirections> imageIds = { 17146, 17147, 17146, 17147 };
```

**Comment by Gymnasiast** [comment]
@tupaschoal 

> I'd rather have us using std::array over plain arrays whenever possible.

This is the kind of thing that really needs discussing first. Let's take this over to #12166.

**Comment by Gymnasiast** [comment]
Because we are still working out the best solution in #12166, I have decided to fix the inconsistent capitalisation and merge this PR, even though I know it will need revisiting. I did this mainly because I wanted to reduce the number of open PRs and because it was still cleaning up code.

@frutiemax Since we're not done discussing #12166 yet, I would recommend you put this refactor on hold until we have decided what it should look like, in order to avoid PRs being open for too long.

**Pull request merged**